### PR TITLE
Add support for POSIX_SPAWN_SETSIGMASK and POSIX_SPAWN_SETSIGDEF

### PR DESCRIPTION
**Pull request by jessehui**

### CODE REVIEW

File: src/libos/src/process/spawn_attribute.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+// name in musl.
+#[repr(C)]
+#[derive(Debug)]
+pub struct posix_spawnattr_t {
+    flags: SpawnAttributeFlags, // __flags
+    pgrp: i32,                  // __pgrp
+    sd: SpawnAttrSigSet,        // __def
+    ss: SpawnAttrSigSet,        // __mask
+    // Below elements are different in musl but we are not using it for now. So just ignore.
+    sp: SchedParam,
```
Comment (guzongmin) [inline src/libos/src/process/spawn_attribute.rs:25]: If we do not use entries, should we add them now? Or can we find a better way to add both glibc and musl implementation?

Comment (jessehui) [reply src/libos/src/process/spawn_attribute.rs]: Remove the elements we don't use for now. Also, adjust the definition order to be clearer.

File: src/libos/src/process/do_clone.rs [hunk 2]
```diff
@@ -59,12 +60,14 @@ pub fn do_clone(
...
             .name(name)
-            .rlimits(rlimits);
+            .rlimits(rlimits)
+            .sig_mask(sig_mask);
         if let Some(ctid) = ctid {
             builder = builder.clear_ctid(ctid);
         }
         builder.build()?
     };
+    warn!("new thread sigmask = {:?}", new_thread_ref.sig_mask());
```
[the remark this answers is not in the document]
Comment (jessehui) [reply src/libos/src/process/do_clone.rs]: Done. Thanks for your review.

**Review by guzongmin — APPROVED** [review]
LGTM

**Pull request merged**